app/assets/javascripts/components/course_queue.js.jsx

```diff
@@ -279,6 +282,16 @@ var CourseQueue = React.createClass({
       </div>
     );
   },
+  pinTopRequest: function () {
+    index = 0;
+    for(var r=0; r< this.state.requests.length; r++){
+      if(!this.state.requests[r].resolver){
+        index = r;
+        break;
+      }
```

Comment:
Is there any reason why `getUnpinnedRequests` wouldn't work here? Something like this maybe?

```js
pinTopRequest: function () {
    let unpinnedRequests = this.getUnpinnedRequests();
    return unpinnedRequests.length > 0 ? this.handler.pin.bind(this.handler, unpinnedRequests[0].id) : null;
}
```